<?xml version="1.0"?>

<launch>

    <!--<node name="rqt_robot_monitor" pkg="rqt_robot_monitor" type="rqt_robot_monitor" />-->
    <!--<node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" />-->
        
    <node name="armadillo_hw_node" pkg="armadillo_hw" type="armadillo_hw_node" respawn="false" output="screen">
        
        <!--roboteq-->
        <param name="load_roboteq_hw" value="true" />
        <param name="left_wheel_joint" value="left_wheel_joint" />
        <param name="right_wheel_joint" value="right_wheel_joint" />
        <param name="roboteq_port" value="/dev/armadillo/ROBOTEQ" />
        <param name="roboteq_baud" value="115200" />
        
        <!--ricboard-->
        <param name="load_ric_hw" value="true" />
        <param name="torso_joint" value="torso_joint" />
        <param name="ric_port" value="/dev/armadillo/RICBOARD" />
        
        <!--battery management system-->
        <param name="load_battery_hw" value="true" />
        <param name="show_warnings" value="false" />
        <param name="low_batt_val" value="20" /><!--low battery warning value (precentage). -1 to disable-->
        <param name="batt_port" value="/dev/armadillo/BMS" />
        
        <!--dxl motors-->
        <param name="load_dxl_hw" value="true" />
        <param name="dxl_port_name" value="/dev/armadillo/USB2DYNAMIXEL" />
        <param name="dxl_port_baudrate" value="1000000" />
        <param name="dxl_protocol" value="2.0" />

    </node>
    
  
    <rosparam file="$(find armadillo_hw)/config/dxl_joints_config.yaml" command="load"/>
    
    <rosparam file="$(find armadillo_hw)/config/dxl_spec_config.yaml" command="load"/>

</launch>
